### PR TITLE
Price Feed Resilience: Audit Fixes

### PR DESCRIPTION
**Pull request by 0xJem**
Fixes from the Guardian audit: 
- H-01: #240
- M-01: #254 
- M-02: #254 
- M-03: #242 
- M-04: #240
- M-05: #254 
- M-06: #238
- M-08: #243 
- M-09: #241 
- M-10: #239 
- M-11: #254 
- M-12: #254 
- M-13: #240
- L-01: #254  
- L-02: #237 
- L-03: #250 
- L-05: #244 
- L-06: #251 
- L-08: #251 
- L-09: #246 
- L-12: #254 
- L-13: #255
- L-16: #254 
- L-19: #247 
- L-20: #245
- I-01: #250 
- I-02: #254 
- I-03: #254 
- I-04: #252 
- I-05: #249 
- I-06: #255
- I-08: #248 
- I-10: #253 
- I-12: #250 
- I-13: #254
- I-14: #255 
- #257 

Also includes:
- Pins foundry to 1.5.1 to avoid changes to the compilation restrictions syntax
- Shards the fork tests to speed up CI

## Summary by CodeRabbit

* **New Features**
  * Added a permissioned pair-based PriceCache, unit-of-account support, non-contract asset metadata, and upgraded price/config flows with factory-level cache wiring.

* **Bug Fixes**
  * Pinned Foundry toolchain to version 1.5.1 across CI.

* **Documentation**
  * Clarified price normalization, non-contract asset handling, feed composition/paths, deviation thresholds, confidence/staleness rules, and upgrade guidance.

* **Tests**
  * Sharded fork tests and greatly expanded test coverage for caching, oracle adapters, feeds, strategies, factories and edge cases.

* **Chores**
  * Updated formatter/npm scripts (enabled caching, refined globs).

### CODE REVIEW

**Comment by 0xJem** [comment]
@coderabbitai review

File: src/test/modules/PRICE.v2/gas.t.sol [hunk 1]
```diff
@@ -10,11 +10,16 @@ contract PriceV2GasTest is PriceV2BaseTest {
...
+        vm.warp(block.timestamp + 1);
+        vm.prank(priceWriter);
+        price.storeObservation(address(onema));
+
+        vm.startSnapshotGas("OlympusPricev2.getPrice.last");
+        (uint256 price_, uint48 timestamp_) = price.getPrice(address(onema), IPRICEv2.Variant.LAST);
         uint256 gasUsed = vm.stopSnapshotGas();
-        assertEq(price_, 2000e18, "weth cached price mismatch");
+        assertEq(price_, 5e18, "onema LAST price mismatch");
+        assertEq(timestamp_, uint48(block.timestamp), "onema LAST timestamp mismatch");
```
[the remark this answers is not in the document]
Comment (0xJem) [reply src/test/modules/PRICE.v2/gas.t.sol:22]: Not needed

File: src/test/modules/PRICE.v2/updateAsset.t.sol [hunk 2]
```diff
@@ -2004,36 +2244,155 @@ contract PriceV2UpdateAssetTest is PriceV2BaseTest {
...
+    }
+
+    function test_whenUpdatingMovingAverage_whenStoreMovingAverageFalse_afterObservationStored_clearsMetadata()
+        public
+        givenAsset_SingleFeed_NoStrategy_StoreMA
+    {
+        vm.warp(block.timestamp + 1);
+
+        vm.prank(priceWriter);
+        price.storeObservation(asset_SingleFeed_NoStrategy_StoreMA);
```
[the remark this answers is not in the document]
Comment (0xJem) [reply src/test/modules/PRICE.v2/updateAsset.t.sol:2943]: Not needed

File: src/test/policies/price/ChainlinkOracleCloneable/latestRoundData.t.sol [hunk 3]
```diff
@@ -46,10 +56,35 @@ contract ChainlinkOracleCloneableLatestRoundDataTest is ChainlinkOracleCloneable
...
+    //  [X] it reverts via SafeCast.toInt256 overflow guard
+
+    function test_whenCachedPairPriceExceedsInt256_reverts() public {
+        uint256 overflowPrice = uint256(type(int256).max) + 1;
+        _setPRICEPrices(address(baseToken), overflowPrice);
+        _setPRICEPrices(address(quoteToken), QUOTE_PRICE);
+        _storePrices();
+
+        vm.expectRevert(bytes("SafeCast: value doesn't fit in an int256"));
+        oracle.latestRoundData();
```
[the remark this answers is not in the document]
Comment (0xJem) [reply src/test/policies/price/ChainlinkOracleCloneable/latestRoundData.t.sol:81]: Not needed

File: src/policies/price/BaseOracleFactory.sol [hunk 4]
```diff
@@ -313,10 +334,25 @@ abstract contract BaseOracleFactory is
...
+    ///             - The factory is disabled
+    ///             - The caller is not admin
+    ///             - `policy_` is zero, not IPriceCache-compatible, or bound to a different Kernel
+    function setPriceCache(
+        address policy_
+    ) external override onlyEnabled onlyAdminRole nonReentrant {
+        // Intentionally does not check `IEnabler(policy_).isEnabled()`: this setter validates
+        // compatibility/kernel only and allows admin-led cache policy rotations regardless of runtime state.
+        _setPriceCache(policy_);
+    }
```
[the remark this answers is not in the document]
Comment (0xJem) [reply src/policies/price/BaseOracleFactory.sol:348]: Not true. `PriceCache.decimals()` is called when adjusting the decimal scale of the result in the oracle deployments.

File: src/policies/price/ERC7726OracleFactory.sol [hunk 5]
```diff
@@ -245,39 +296,96 @@ contract ERC7726OracleFactory is
...
+        if (enableData_.length == 0) return;
+
+        (address[] memory baseTokens, address[] memory quoteTokens) = abi.decode(
+            enableData_,
+            (address[], address[])
+        );
+        uint256 pairCount = baseTokens.length;
+        if (pairCount != quoteTokens.length) {
+            revert ERC7726OracleFactory_InvalidEnableData(pairCount, quoteTokens.length);
+        }
```
[the remark this answers is not in the document]
Comment (0xJem) [reply src/policies/price/ERC7726OracleFactory.sol:379]: The enable data isn't fixed length (it's an array with an arbitrary number of elements), so it's difficult to check this. Blank revert is OK.

File: src/test/modules/PRICE.v2/unregisterNonContractAsset.t.sol [hunk 6]
```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: Unlicense
+/// forge-lint: disable-start(mixed-case-variable,mixed-case-function)
+pragma solidity >=0.8.0;
```
[the remark this answers is not in the document]
Comment (0xJem) [reply src/test/modules/PRICE.v2/unregisterNonContractAsset.t.sol]: Not possible to require 0.8.24 due to other dependencies, but 0.8.15 is possible.
